dirsrvtests/tests/suites/schema/schema_csn_replication_test.py

```diff
@@ -0,0 +1,386 @@
+
+    :id: 8a9b5c42-6d7e-4f3c-a5b2-1e9f8d7c6b5a
+    :setup: Two supplier replication topology with test users
+    :steps:
+        1. Add custom attribute types and object classes to supplier2
+        2. Modify standard schema definitions on supplier2
+        3. Trigger replication from supplier1 to supplier2
+        4. Verify schema CSN synchronization
+    :expectedresults:
+        1. Schema changes are applied successfully
+        2. Replication works correctly
+        3. Schema CSNs are synchronized between suppliers
+    """
+    supplier1 = topology_m2.ms["supplier1"]
+    supplier2 = topology_m2.ms["supplier2"]
+    test_users = schema_csn_setup
+    test_user = test_users[0]
+    repl = ReplicationManager(DEFAULT_SUFFIX)
+
+    log.info('Add custom attribute types and object classes to supplier2')
+    schema2 = Schema(supplier2)
+
+    custom_at = create_custom_attributetype('testAttr', 1)
+    schema2.add_attributetype(custom_at)
+
+    custom_oc = create_custom_objectclass('testClass', 1)
+    schema2.add_objectclass(custom_oc)
+
+    log.info('Trigger replication from supplier1 to supplier2')
+    trigger_replication(repl, supplier2, supplier1, test_user)
+
+    users1 = UserAccounts(supplier1, DEFAULT_SUFFIX)
+    user1 = users1.get(test_user.get_attr_val_utf8('uid'))
+    assert user1.get_attr_val_utf8('description') == test_user.get_attr_val_utf8('description')
+
+    log.info('Verify schema CSN synchronization')
+    wait_for_schema_csn_sync(supplier1, supplier2)
+
+
+def test_schema_csn_after_data_replication(topology_m2, schema_csn_setup):
+    """Test that schema CSN synchronizes after normal data replication
+
+    :id: 7f8e9d6c-5b4a-3c2d-1e0f-9e8d7c6b5a49
+    :setup: Two supplier replication topology with existing schema changes
+    :steps:
+        1. Perform a data modification on supplier1
+        2. Wait for replication to complete
+        3. Check schema CSN synchronization with longer wait if needed
+    :expectedresults:
+        1. Data replication works correctly
+        2. Schema CSNs eventually synchronize between suppliers
+    """
+    supplier1 = topology_m2.ms["supplier1"]
+    supplier2 = topology_m2.ms["supplier2"]
+    test_users = schema_csn_setup
+    test_user = test_users[1]
+    repl = ReplicationManager(DEFAULT_SUFFIX)
+
+    log.info('Perform a data modification on supplier1')
+    trigger_replication(repl, supplier1, supplier2, test_user)
+
+    users2 = UserAccounts(supplier2, DEFAULT_SUFFIX)
+    user2 = users2.get(test_user.get_attr_val_utf8('uid'))
+    assert user2.get_attr_val_utf8('description') == test_user.get_attr_val_utf8('description')
+
+    log.info('Check schema CSN synchronization with longer wait if needed')
+    wait_for_schema_csn_sync(supplier1, supplier2)
+
+
+def test_schema_learning_with_paused_agreement(topology_m2, schema_csn_setup, request):
+    """Test schema learning when replication agreement is paused
```

Comment:
I think the original test was written before the introduction of the keep alive entry that is periodic. If supplier2 creates/updates its keep alive entry by the time we check schemaCSN there is a possibility that this test will fail.
IMHO I think we can abandon this test.

Reply:
Keep alive entry is added/updated 30sec after startup. If the test is launched far from that delay then it is safe keeping the test. You may also restart the instance at the beginning of the tests